gradio/blocks.py

```diff
@@ -12,7 +12,9 @@
 
 
 class Block:
-    def __init__(self):
+    def __init__(self, without_rendering=False):
```

Comment:
is it possible that instead of using "without_rendering", it just checks to see if its within a BlockContext and only calls render if that's the case?

Reply:
That doesn't work. Consider the following example:

```python
with gr.Blocks() as bl:
    with gr.Tabs():
        with gr.TabItem("Text"):
            gr.Interface(lambda x:x, "textbox", "textbox")
        with gr.TabItem("Image"):
            gr.Interface(lambda x:x, "image", "image")
    bl.launch()
```

Inside the `Interface` class, you create `Component` objects once, and then render them later. 

Create components: https://github.com/gradio-app/gradio/blob/5c44ae353672026984df7cdb9157ff9fc6cb1d4a/gradio/interface.py#L182

Render: https://github.com/gradio-app/gradio/blob/5c44ae353672026984df7cdb9157ff9fc6cb1d4a/gradio/interface.py#L496

In this case, both statements are inside a `BlockContext` (the outer level `Blocks` `bl`), so the component would end up getting rendered twice. That's why we need to explicitly escape the context.